MatiasDure/Project_IMTS
# Chest inspection active interaction

**Description**
* The user interacts with the chest object through the portal and a chest inspection event starts with the following activities:
1. The chest opens
2. The bee goes to the chest
3. the user can interact with the chest again
4. If so, the bee goes into the chest
5. The chest closes
6. A delay starts and the user can interact with the chest again
7. After the delay ends or the user interacts with the chest again, the chest opens and the bee is free
* Interruptions works as expected.

**Screenshot**
![ChestInspection](https://github.com/user-attachments/assets/976d7713-0216-4016-b810-5400abaee02c)

**How To Test**
1. Switch to the `chest-inspection-active-interaction` branch
2. Open the unity project
3. Open the `ChestInspectionActiveInteraction` scene
4. Run the scene
5. Find the portal (Should be on the right)
6. There should be two chests inside the portal
7. Click any chests and try to break it
